server/src/com/thoughtworks/go/server/service/PipelineConfigsService.java

```diff
@@ -113,4 +116,17 @@ private boolean userHasPermissions(Username username, String groupName, HttpLoca
         return LocalizedMessage.string("UPDATE_GROUP_XML_FAILED", groupName, message);
     }
 
+    public void updateAuthorization(Authorization authorization, String groupName, HttpLocalizedOperationResult result, Username validUser) {
+        UpdateAuthorizationCommand command = new UpdateAuthorizationCommand(groupName, authorization);
+        try{
+            PipelineConfigsUpdateCommand updateCommand = new PipelineConfigsUpdateCommand(goConfigService, command, result, validUser, groupName);
+            goConfigService.updateConfig(updateCommand, validUser);
+        } catch (Exception e) {
+            if (e instanceof GoConfigInvalidException) {
+                result.unprocessableEntity(LocalizedMessage.string("ENTITY_CONFIG_VALIDATION_FAILED"));
+            } else {
+                result.internalServerError(LocalizedMessage.string("INTERNAL_SERVER_ERROR"));
```

Comment:
can we also show the error along with the message
`result.internalServerError(LocalizedMessage.string("INTERNAL_SERVER_ERROR", e.getMessage()));`